logstash-core/src/main/java/org/logstash/instrument/metrics/UptimeMetric.java

```diff
@@ -0,0 +1,83 @@
+package org.logstash.instrument.metrics;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.Objects;
+
+/**
+ * The {@link UptimeMetric} is an auto-advancing {@link Metric} whose value
+ * represents the amount of time since instantiation. It can be made to track the
+ * advancement of the clock in one of several {@link TemporalUnit}s.
+ */
+public class UptimeMetric extends AbstractMetric<Long> {
+    private final Clock clock;
+    private final Instant startInstant;
```

Comment:
This should likely be implemented in terms of `System#nanoTime`, not `Clock`/`Instant`, since we are tracking _elapsed_ time and do not need to track against the UTC timeline (which has surprises like leap-seconds).